pkg/controller/ephemeraljob/ephemeraljob_utils.go

```diff
@@ -151,7 +168,17 @@ func getSyncPods(job *appsv1alpha1.EphemeralJob, pods []*v1.Pod) (toCreate, toUp
 func calculateEphemeralContainerStatus(job *appsv1alpha1.EphemeralJob, pods []*v1.Pod) error {
 	var success, failed, running, waiting int32
 	for _, pod := range pods {
-		state, err := parseEphemeralPodStatus(job, econtainer.New(job).GetEphemeralContainersStatus(pod))
+		eContainerStatus, err := econtainer.New(job).GetEphemeralContainersStatus(pod)
```

Comment:
Put `ecImpl:=econtainer.New(job)` before loop and use `ecImpl.GetEphemeralContainersStatus(pod)` for each pod.